Game/Scripts/ActivationLogic.cpp

```diff
@@ -53,21 +57,26 @@ void ActivationLogic::Update(float deltaTime)
 
 void ActivationLogic::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
 		componentAnimation->SetParameter("IsActive", true);
 		componentRigidBody->Disable();
 		componentAudio->PostEvent(AUDIO::SFX::AMBIENT::SEWERS::BIGDOOR_OPEN);
 	}
+	catch(const ComponentNotFoundException){
```

Comment:
From what I gathered, the C++ standard is to catch by reference. Please change this catch clause and the one in line 80.
```suggestion
	catch(const ComponentNotFoundException&){
```